doc/tutorials/klondike/app/lib/step3/components/card.dart

```diff
@@ -0,0 +1,209 @@
+        _drawSprite(canvas, suitSprite, 0.3, 0.2);
+        _drawSprite(canvas, suitSprite, 0.7, 0.2);
+        _drawSprite(canvas, suitSprite, 0.5, 0.35);
+        _drawSprite(canvas, suitSprite, 0.3, 0.5);
+        _drawSprite(canvas, suitSprite, 0.7, 0.5);
+        _drawSprite(canvas, suitSprite, 0.3, 0.2, rotate: true);
+        _drawSprite(canvas, suitSprite, 0.7, 0.2, rotate: true);
+        break;
+      case 8:
+        _drawSprite(canvas, suitSprite, 0.3, 0.2);
+        _drawSprite(canvas, suitSprite, 0.7, 0.2);
+        _drawSprite(canvas, suitSprite, 0.5, 0.35);
+        _drawSprite(canvas, suitSprite, 0.3, 0.5);
+        _drawSprite(canvas, suitSprite, 0.7, 0.5);
+        _drawSprite(canvas, suitSprite, 0.3, 0.2, rotate: true);
+        _drawSprite(canvas, suitSprite, 0.7, 0.2, rotate: true);
+        _drawSprite(canvas, suitSprite, 0.5, 0.35, rotate: true);
+        break;
+      case 9:
+        _drawSprite(canvas, suitSprite, 0.3, 0.2);
+        _drawSprite(canvas, suitSprite, 0.7, 0.2);
+        _drawSprite(canvas, suitSprite, 0.5, 0.3);
+        _drawSprite(canvas, suitSprite, 0.3, 0.4);
+        _drawSprite(canvas, suitSprite, 0.7, 0.4);
+        _drawSprite(canvas, suitSprite, 0.3, 0.2, rotate: true);
+        _drawSprite(canvas, suitSprite, 0.7, 0.2, rotate: true);
+        _drawSprite(canvas, suitSprite, 0.3, 0.4, rotate: true);
+        _drawSprite(canvas, suitSprite, 0.7, 0.4, rotate: true);
+        break;
+      case 10:
+        _drawSprite(canvas, suitSprite, 0.3, 0.2);
+        _drawSprite(canvas, suitSprite, 0.7, 0.2);
+        _drawSprite(canvas, suitSprite, 0.5, 0.3);
+        _drawSprite(canvas, suitSprite, 0.3, 0.4);
+        _drawSprite(canvas, suitSprite, 0.7, 0.4);
+        _drawSprite(canvas, suitSprite, 0.3, 0.2, rotate: true);
+        _drawSprite(canvas, suitSprite, 0.7, 0.2, rotate: true);
+        _drawSprite(canvas, suitSprite, 0.5, 0.3, rotate: true);
+        _drawSprite(canvas, suitSprite, 0.3, 0.4, rotate: true);
+        _drawSprite(canvas, suitSprite, 0.7, 0.4, rotate: true);
+        break;
+      case 11:
+        _drawSprite(canvas, suit.isRed ? redJack : blackJack, 0.5, 0.5);
+        break;
+      case 12:
+        _drawSprite(canvas, suit.isRed ? redQueen : blackQueen, 0.5, 0.5);
+        break;
+      case 13:
+        _drawSprite(canvas, suit.isRed ? redKing : blackKing, 0.5, 0.5);
+        break;
+    }
+  }
+
+  void _drawSprite(
+    Canvas canvas,
+    Sprite sprite,
+    double relativeX,
+    double relativeY, {
+    double scale = 1,
+    bool rotate = false,
+  }) {
+    if (rotate) {
+      canvas.save();
+      canvas.translate(size.x / 2, size.y / 2);
+      canvas.rotate(pi);
+      canvas.translate(-size.x / 2, -size.y / 2);
+    }
+    sprite.render(
+      canvas,
+      position: Vector2(relativeX * size.x, relativeY * size.y),
+      anchor: Anchor.center,
+      size: sprite.srcSize.scaled(scale),
+    );
+    if (rotate) {
+      canvas.restore();
+    }
+  }
```

Comment:
I guess, it's easier for me to think of a `Card` as a single indecomposable physical object: just a piece of hard paper with multiple images drawn on its surface.

Of course, it would have been possible to model `Card` as a group, where each small image is its own component, but I feel like it wouldn't make the code any simpler. If there was ever a need to move those images on the surface of a card, or apply effects to them, then using components would have been sensible; but I do not anticipate any such need. In this case it seems easier to have just a single `Card` component that draws several sprites manually.

Reply:
I agree that it's a bit of an advanced topic, but if the reader has difficulty understanding it -- they can safely skip this part without sacrificing their understanding of the rest of the material.

On the other hand, going with `SpriteComponent`s here would add extra 440 components into the game, which would also be challenging for the user to understand.

A kind of a compromise would have been if we added `rotate` argument directly into `Sprite.render()` -- what do you think?

Reply:
> A kind of a compromise would have been if we added `rotate` argument directly into `Sprite.render()` -- what do you think?

That's an interesting idea! Especially since it already has an `anchor`.

I checked with the team and they are onboard with the idea, but it should be done as a separate PR.

Reply:
:+1: